File: Android/app/src/main/java/com/dkaluta/prosary/models/JesusPrayerOptions.kt
package com.dkaluta.prosary.models

import android.content.Context
import com.dkaluta.prosary.R

/** Configuration options specific to the Jesus Prayer. Lives inside a [Prayer] when kind == JesusPrayer. */
data class JesusPrayerOptions(
    var target: JesusPrayerTarget = JesusPrayerTarget.Count(33),
) {
    fun targetDisplayName(context: Context): String = when (val t = target) {
        is JesusPrayerTarget.Count -> context.getString(R.string.jp_count_times, t.value)
        JesusPrayerTarget.Unbounded -> context.getString(R.string.jp_unbounded)
    }
}

File: Windows/Prosary/Models/JesusPrayerOptions.cs
using Prosary.Localization;

namespace Prosary.Models;

/// <summary>Configuration options specific to the Jesus Prayer. Lives inside a <see cref="Prayer"/>
/// when <c>Kind == PrayerKind.JesusPrayer</c>.</summary>
public sealed record JesusPrayerOptions
{
    public JesusPrayerTarget Target { get; init; } = new JesusPrayerTarget.Count(33);

    public string TargetDisplayName => Target switch
    {
        JesusPrayerTarget.Count(var n) => string.Format(Loc.Tr("jp_count_times", "{0}×"), n),
        JesusPrayerTarget.Unbounded => Loc.Tr("jp_unbounded", "Unbounded"),
        _ => throw new ArgumentOutOfRangeException(nameof(Target))
    };
}
